command/agent/agent_endpoint.go

```diff
@@ -199,6 +204,11 @@ func (s *HTTPServer) AgentRegisterService(resp http.ResponseWriter, req *http.Re
 		}
 	}
 
+	// Get the provided token, if any
+	var token string
+	s.parseToken(req, &token)
+	s.agent.state.SetServiceToken(ns.ID, token)
```

Comment:
I feel like we should just add this to the `AddService` call instead of reaching into `state` directly